src/app/routes/search/search.component.ts

```diff
@@ -2,36 +2,86 @@ import { Subscription } from 'rxjs';
 import { Component, OnInit } from '@angular/core';
 import { FormsModule } from '@angular/forms';
 import { ActivatedRoute, Params } from '@angular/router';
+import { CURRENT_YEAR } from '../../config';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/debounceTime';
+import 'rxjs/add/operator/distinctUntilChanged';
 
+import { RequestService } from '../../RequestService/request.service';
 import { SearchResultsComponent } from '../../shared/shared';
+import { SearchableFields } from '../../shared/fields';
 
-// @Component({
-//   template:  `
-    // <div class="container">
-    //   <h2>Search</h2>
-    //   <!--Two way data binding is cool but not very efficient in this case. -->
-    //   <input [(ngModel)]='query'>
-    //   <search-results [query]='query'></search-results>
-    // </div>
-//   `,
-// })
 
 @Component({
   templateUrl: 'search.component.html',
   styleUrls: ['search.component.css'],
-}) 
+  providers: [ RequestService ],
+})
 
 export class SearchComponent implements OnInit {
-  query: String;
-  private subscription: Subscription;
+  typedQuery: string;
+  searchQuery: string;
+  allProfiles: any[] = [];
+  typeaheadResults: string[] = [];
+  resultsSub: Subscription;
+  typeaheadSub: Subscription;
 
-  constructor(private activatedRoute: ActivatedRoute) {}
+  constructor(private activatedRoute: ActivatedRoute, private rs: RequestService) {}
 
   ngOnInit() {
     // subscribe to router event
-    this.subscription = this.activatedRoute.params.subscribe(
+    this.resultsSub = this.activatedRoute.params.subscribe(
       (param: any) => {
-        this.query = param['query'];
+        this.searchQuery = param['query'];
     });
+    // get all profile view info
+    var query = this.typedQuery || "";
+    this.typeaheadSub = this.rs.getWithSub('/search/'+ CURRENT_YEAR + "/" + query , (data) => {
+      this.allProfiles = data.results.sort((p1,p2) => {
```

Comment:
Server now responds with data already sorted. 🙌 So we shouldn't need this. (I'll look into it).